libcaf_core/caf/logger.cpp

```diff
@@ -40,6 +40,177 @@ thread_local actor_id current_actor_id;
 // Stores a pointer to the system-wide logger.
 thread_local intrusive_ptr<logger> current_logger_ptr;
 
+// Default logger implementation.
+class default_logger : public logger, detail::atomic_ref_counted {
```

Comment:
```suggestion
// Default logger implementation.
class default_logger : public logger, public detail::atomic_ref_counted {
```

Also, this class is in the anonymous namespace but the declaration is outside of the namespace. In any case, there is no need to implement the member functions out-of-line anymore. Please move the implementations to the class definition.